analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,241 @@
+
+# run per sample on standard fusion calls
+standardFusioncallsPerSample<-lapply(as.character(sampleList),function(x) fusion_filtering_expression(Sample = x,expressionFilter = expressionFilter,standardFusioncalls = QCFiltered))
+
+QCExpFilteredstandardFusioncalls <- Reduce(function(x, y) rbind (x,y), standardFusioncallsPerSample)
+saveRDS(QCExpFilteredstandardFusioncalls,paste0(opt$outputfile,"_QC_expression_filtered.RDS"))
+
+#####################################
+
+
+###############annotation############
+# column 1 as GeneName 2 source file 3 Type; collapse to summarize type
+geneListReferenceDataTab<-read.delim(paste0(referenceFolder,"genelistreference.txt"),stringsAsFactors = FALSE)
+geneListReferenceDataTab<-geneListReferenceDataTab %>% group_by(Gene_Symbol) %>% mutate(type = toString(type)) %>% as.data.frame()
+geneListReferenceDataTab<-unique(geneListReferenceDataTab[,c("Gene_Symbol","type")])
+
+# column 1 as FusionName 2 source file 3 Type; collapse to summarize type
+fusionReferenceDataTab<-read.delim(paste0(referenceFolder,"fusionreference.txt"),stringsAsFactors = FALSE)
+fusionReferenceDataTab<-unique(fusionReferenceDataTab[,c("FusionName","type")])
+
+fusion_annotation_list<-function(standardFusioncalls=standardFusioncalls,geneListReference=geneListReference,fusionReference=fusionReference){
+  # @param standardFusioncalls A dataframe from star fusion or arriba standardized and QC filtered 
+  # @param geneListReference A dataframe of genes of interest ; columns 1 : GeneNames 2: Source file 3: Type
+  # @param fusionReference A dataframe of fusion of interest ; columns 1 : FusionName 2: Source file 3: Type
+  # @return Standardized and QC filtered fusion calls annotated with gene list and fusion calls from reference lists
+  
+#Gene annotation
+standardFusioncallsGOI<-standardFusioncalls %>% 
+           mutate(Gene1A_anno=lapply(standardFusioncalls$Gene1A,function(x) geneListReferenceDataTab[which(geneListReferenceDataTab$Gene_Symbol==x),"type"]),
+           Gene2A_anno=lapply(standardFusioncalls$Gene2A,function(x) geneListReferenceDataTab[which(geneListReferenceDataTab$Gene_Symbol==x),"type"]),
+           Gene1B_anno=lapply(standardFusioncalls$Gene1B,function(x) geneListReferenceDataTab[which(geneListReferenceDataTab$Gene_Symbol==x),"type"]),
+           Gene2B_anno=lapply(standardFusioncalls$Gene2B,function(x) geneListReferenceDataTab[which(geneListReferenceDataTab$Gene_Symbol==x),"type"]))
+  
+#Fusion annotation  
+standardFusioncallsFOI<-standardFusioncalls %>% 
+  mutate(Fusion_anno=lapply(standardFusioncalls$FusionName,function(x) fusionReferenceDataTab[which(fusionReferenceDataTab$FusionName==x),"type"]))
+
+standardFusioncalls<-list("geneAnnotion"=standardFusioncallsGOI,"fusionAnnottaion"=standardFusioncallsFOI)
+
+return(standardFusioncalls)
+}
+
+
+standardFusioncallsAnnotated<-fusion_annotation_list(standardFusioncalls = standardFusioncalls,geneListReference=geneListReference,fusionReference=fusionReference)
```

Comment:
Yes you are right it should be geneListReferenceDataTab, I changed every occurence of geneListReference to geneListReferenceDataTab because then only the function param declaration needed changes